app/uk/gov/hmrc/customs/declaration/services/StatusResponseValidationService.scala

```diff
@@ -27,29 +28,110 @@ import scala.xml.NodeSeq
+    })(acceptanceDate => {
+      val parsedDateTime = ISODateTimeFormat.dateTimeNoMillis.withZoneUTC().parseDateTime(acceptanceDate.head)
+      val isDateValid = parsedDateTime.isAfter(getValidDateTimeUsingConfig)
+      if (!isDateValid) {
+        declarationsLogger.debugWithoutRequestContext(s"Status response acceptanceDate failed validation $acceptanceDate")
+        Left(ErrorResponse.errorBadRequest(s"Declaration acceptance date is greater than ${declarationsConfigService.declarationsConfig.declarationStatusRequestDaysLimit} days old"))
+      } else{
+        Right(true)
+      }
+    })
   }
 
+  def extractTradeMovementType(declarationNode: NodeSeq): Either[ErrorResponse, String]  = {
+    val maybeExtractedTradeMovementType = extractField(declarationNode, "tradeMovementType").fold[Option[String]](None)(tradeMovementType => Some(tradeMovementType.head.substring(0,2)))
+
+    maybeExtractedTradeMovementType.fold[Either[ErrorResponse, String]]({
+      Right(IMPORT_MOVEMENT_TYPE)
+    })(extractedTradeMovementType => {
+      if(extractedTradeMovementType.equals(CO_MOVEMENT_TYPE)) {
+        Right(deriveTradeMovementTypeFromProcedureCategory(extractProcedureCategory(declarationNode)))
+      } else{
+        Right(extractedTradeMovementType)
+      }
+    })
+  }
 
-  private def validateReceivedDate(declarationNode: NodeSeq): Boolean = {
-    extractField(declarationNode, "receiveDate").fold(false)(receiveDate => {
-     val parsedDateTime = ISODateTimeFormat.dateTime().withZone(DateTimeZone.UTC).parseDateTime(receiveDate.head)
-     val isDateValid = parsedDateTime.isAfter(getValidDateTimeUsingConfig)
-     if (!isDateValid) declarationsLogger.debugWithoutRequestContext(s"Status response receivedDate failed validation $receiveDate")
-     isDateValid
-   })
+  private def deriveTradeMovementTypeFromProcedureCategory(maybeProcedureCategory: Option[String]):  String  = {
+    if(isImportProcedureCategory(maybeProcedureCategory)){
+      IMPORT_MOVEMENT_TYPE
+    } else if(isExportProcedureCategory(maybeProcedureCategory)){
+      EXPORT_MOVEMENT_TYPE
+    } else{
+      IMPORT_MOVEMENT_TYPE
+    }
+  }
+
+  private def isExportProcedureCategory(maybeProcedureCategory: Option[String]): Boolean = {
+    matchAgainstGivenCategories(maybeProcedureCategory, exportProcedureCategories)
+  }
+
+  private def isImportProcedureCategory(maybeProcedureCategory: Option[String]): Boolean = {
+    matchAgainstGivenCategories(maybeProcedureCategory, importProcedureCategories)
+  }
+
+  private def matchAgainstGivenCategories(maybeProcedureCategory: Option[String], categoriesToMatch: Seq[String]): Boolean ={
+    maybeProcedureCategory.exists(procedureCategory => categoriesToMatch.contains(procedureCategory))
+  }
+
+  private def extractProcedureCategory(declarationNode: NodeSeq): Option[String]  = {
+    extractField(declarationNode, "procedureCategory").fold[Option[String]](None)(procedureCategory => Some(procedureCategory.head.substring(0,2)))
```

Comment:
as above, how bomb-proof do we want to make the data extractions?

Reply:
Think you mean the empty or 1 character edge cases. I'll cover this off when I add add tests and function to protect against index out of bounds etc.